handlers/add_teacher.go

```diff
@@ -0,0 +1,37 @@
+package handlers
+
+import (
+	"fmt"
+
+	runtime "github.com/ahab94/ApnaSchool"
+	"github.com/ahab94/ApnaSchool/gen/restapi/operations"
+	"github.com/ahab94/ApnaSchool/models"
+	"github.com/go-openapi/runtime/middleware"
+)
+
+// NewAddTeacher handles request for saving teacher
+func NewAddTeacher(rt *runtime.Runtime) operations.AddTeacherHandler {
+	return &addTeacher{rt: rt}
+}
+
+type addTeacher struct {
+	rt *runtime.Runtime
+}
+
+// Handle the add teacher request
+func (d *addTeacher) Handle(params operations.AddTeacherParams) middleware.Responder {
+	teacher := models.Teacher{
+		ID:         params.Teacher.ID,
```

Comment:
We shouldn't get id in the json. Passing an ID in post call should be restricted.